### ISSUE TITLE
Preview widens when replacing `100%` by `parent.width`

### ISSUE THREAD

**Issue by Enyium** [slint-ui/slint#6321]
Consider [this SlintPad demo](https://slintpad.com/?snippet=import+%7B+HorizontalBox+%7D+from+%22std-widgets.slint%22%3B%0A%0Aexport+component+Demo+%7B%0A++++HorizontalBox+%7B%0A++++++++alignment%3A+center%3B%0A++++++++%2F%2Fpadding%3A+0%3B+%2F%2F+Bug+doesn%27t+happen+with+this.%0A%0A++++++++Rectangle+%7B%0A++++++++++++%2F%2F+Every+cutting+of+%60100%25%3B%2F%2F%60+makes+preview+widen.%0A++++++++++++width%3A+100%25%3B%2F%2Fparent.width%3B%0A%0A++++++++++++height%3A+20px%3B%0A++++++++++++background%3A+fuchsia%3B%0A++++++++%7D%0A++++%7D%0A%7D%0A&style=native) (see comments):

```slint
import { HorizontalBox } from "std-widgets.slint";

export component Demo {
    HorizontalBox {
        alignment: center;
        //padding: 0; // Bug doesn't happen with this.

        Rectangle {
            // Every cutting of `100%;//` makes preview widen.
            width: 100%;//parent.width;

            height: 20px;
            background: fuchsia;
        }
    }
}
```

Happens in SlintPad and VS Code preview window.

**Comment by ogoffart** [comment]
Duplicate of https://github.com/slint-ui/slint/issues/2902

(percentages size are considered differently by the layout than a literal parent.width.
There is, in fact, a binding loop there but we don't detect it, see #2902)
